tests/functional/fx_fennec_v1_sign_up.js

```diff
@@ -58,6 +58,20 @@ define([
         })
 
         .then(FunctionalHelpers.testIsBrowserNotified(self, 'fxaccounts:can_link_account'))
```

Comment:
> I remember we talked about this and there is an issue with getting into weird states if account is created before the choice made. This should happen after the user submits the choose what to Sync page

I don't quite understand, the account is not created when the `fxaccounts:can_link_account` message is sent. The desktop browser (not sure how Fennec operates) just checks to see if the hash of the email address is the same as the one that is already stored. If not, ask the user if they want to join data.